tests/unit/test_protocol_errors.py

```diff
@@ -0,0 +1,105 @@
+"""
```

Comment:
Good tests, but it strikes me that these should be in a compliance suite.  Of course, we don't have one yet, so this is fine for the time being, but once we get one, we should consider whether moving these there (or duplicating them in the client suite) would be the better option...